docker/solr/entrypoint.sh

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env bash
+
+SOLR_PORT=${SOLR_PORT:-8983}
+SOLR_VERSION=${SOLR_VERSION:-4.9.1}
+DEBUG=${DEBUG:-false}
+SOLR_CORE=${SOLR_CORE:-core0}
+# Since Solr 5.x
+SOLR_COLLECTION=${SOLR_COLLECTION:-gettingstarted}
+
+download() {
+    FILE="$2.tgz"
+    if [ -f $FILE ];
+    then
+       echo "File $FILE exists."
+       tar -zxf $FILE
+    else
+       echo "File $FILE does not exist. Downloading solr from $1..."
+       curl -O $1
+       tar -zxf $FILE
+    fi
+    echo "Downloaded!"
+}
+
+is_solr_up(){
+    echo "Checking if solr is up on http://localhost:$SOLR_PORT/solr/admin/cores"
+    http_code=`echo $(curl -s -o /dev/null -w "%{http_code}" "http://localhost:$SOLR_PORT/solr/admin/cores")`
+    return `test $http_code = "200"`
+}
+
+wait_for_solr(){
+    while ! is_solr_up; do
+        sleep 3
+    done
+}
+
+keep_solr_up(){
+    trap "exit" INT
+    while is_solr_up; do
+        sl -e
+        echo "Still up!" 
+        sleep 2
+    done
+}
+
+run() {
+    dir_name=$1
+    solr_port=$2
+    solr_core=$3
+    # Run solr
+    echo "Running with folder $dir_name"
+    echo "Starting solr on port ${solr_port}..."
+
+    # go to the solr folder
+    cd $1/example
+
+    if [ "$DEBUG" = "true" ]
+    then
+        java -Djetty.port=$solr_port -Dsolr.solr.home=multicore -jar start.jar &
+    else
+        java -Djetty.port=$solr_port -Dsolr.solr.home=multicore -jar start.jar > /dev/null 2>&1 &
+    fi
+    wait_for_solr
+    cd ../../
+    echo "Started"
+}
+
+run_solr5_example() {
+    dir_name=$1
+    solr_port=$2
+    ./$dir_name/bin/solr -p $solr_port -c -e schemaless
+    echo "Started"
+}
+
+run_solr5() {
+    dir_name=$1
+    solr_port=$2
+    ./$dir_name/bin/solr -p $solr_port -c
+    echo "Started"
+}
+
+download_and_run() {
```

Comment:
Downloading Solr could be done at build-time rather than at start time (and we don't need the support of multiple Solr versions, as Packagist wants to use only 3.6, and if it migrates to Solr 6 in the future, it should also migrate to the official image)

Reply:
Done!